src/fileop.c

```diff
@@ -196,3 +255,195 @@ int librole_get_directory_files(const char * const directory,
 librole_get_directory_files_end:
     return retcode;
 }
+
+/**
+ * \brief Read system role entries from /etc/role.d catalog.
```

Comment:
I suggest moving this documentation to header file because the function is "public" (declared in header file).